core/src/main/java/hudson/scm/SCMDescriptor.java

```diff
@@ -56,7 +58,7 @@
      * @deprecated No longer used by default.
      */
     @Deprecated
-    public volatile int generation = 1;
+    public volatile AtomicInteger generation = new AtomicInteger(1);
```

Comment:
In theory this is public API available for use in plugins, and a breaking change if any of them do. Have you checked whether this is used in any plugins e.g. using https://github.com/jenkins-infra/usage-in-plugins ?

Reply:
Sorry, I should have mentioned, that I found it here: https://ci.jenkins.io/job/Infra/job/usage-in-plugins/job/master/lastSuccessfulBuild/artifact/output/deprecated-and-unused.html

Reply:
In that case you should add `@Restricted(NoExternalUse.class) @RestrictedSince("TODO")` as well, to prevent future uses.

Reply:
Thanks, I added this.